MarcusBarnes/mik
# Issue 370

**Github issue**: (#370)

# What does this Pull Request do?

Provides a configuration option for the CSV books and newspapers toolchains that allows the use of a page-sequence delimiter other than the currently hard-coded `-`.

# What's new?

This PR adds a new configuration option used by the CSV books and newspapers writers to define the page sequence delimiter, defaulting to a hyphen (`-`), so this change should not be backwards-breaking. If a user wants to use another. The delimter is defined by the user by adding the following setting in the `[WRITER]` section of the .ini file:

```
page_sequence_separator = _
```

for example.

# How should this be tested?
Following the new "testable" workflow originally described in #370 and under discussion in #384:

1. Check out the issue-370 branch.
1. Run `phpunit --exclude-group inputvalidators --bootstrap vendor/autoload.php tests`.
1. All tests should pass.

The PHPUnit tests add a non-default value to the XCSVBooks test suite's `testWritePackages()` test, and provides assets that are also used in that test. The default case (where the delimeter uses the default of `-`) is coverd by the XCsvNewspapers tests (which still pass) are unaltered in this PR.
